src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -566,4 +568,13 @@ private static void validateSingleQueryType(Integer k, Float distance, Float sco
             throw new IllegalArgumentException(String.format("[%s] requires exactly one of k, distance or score to be set", NAME));
         }
     }
+
+    private static <T> void updateQueryStats(T queryParam, QueryBuilder filter, KNNCounter queryCounter, KNNCounter queryWithFilterCounter) {
+        if (queryParam != null) {
+            queryCounter.increment();
+            if (filter != null) {
+                queryWithFilterCounter.increment();
+            }
+        }
+    }
```

Comment:
This is an interesting choice of abstraction. Would it be better if we have just created a list of KNNCounters that needs to be incremented, and then just pass on that list to a function may be incrementCounters(List<KNNCounters> countersToBeIncremented) of KNNCounter. I think that would be better simplified the logic for below reasons:
1. You will avoid extra null checks for every queryParam.
2. You will avoid null checks on the filter param. 
Because next time if a new query type will come we will need to repeat below thing for it too.

```
 updateQueryStats(maxDistance, filter, KNNCounter.MAX_DISTANCE_QUERY_REQUESTS, KNNCounter.MAX_DISTANCE_QUERY_WITH_FILTER_REQUESTS);
```

which not required.

Reply:
@navneet1v Do you mean use it like below? it could simplify the logic, but I have light concern about the call parameter of countersToBeIncremented list order, we must be careful to prepare this parameter order to use.

```
    private static <T> void updateQueryStats(T queryParam, QueryBuilder filter, List<KNNCounter> countersToBeIncremented) {
        if (queryParam != null) {
            countersToBeIncremented.get(0).increment();
            if (filter != null) {
                countersToBeIncremented.get(1).increment();
            }
        }
    }

updateQueryStats(k, filter, List.of(KNNCounter.KNN_QUERY_REQUESTS, KNNCounter.KNN_QUERY_WITH_FILTER_REQUESTS));
updateQueryStats(minScore, filter, List.of(KNNCounter.MIN_SCORE_QUERY_REQUESTS, KNNCounter.MIN_SCORE_QUERY_WITH_FILTER_REQUESTS));
updateQueryStats(maxDistance, filter, List.of(KNNCounter.MAX_DISTANCE_QUERY_REQUESTS, KNNCounter.MAX_DISTANCE_QUERY_WITH_FILTER_REQUESTS));
```